app/controllers/stats_controller.rb

```diff
@@ -0,0 +1,25 @@
+class StatsController < ApplicationController
+  before_action :authorize_access
+
+  def new
+    @model = StatRange.new
+  end
+
+  def create
+    from = DateFieldCollection.new(params.require(:stat_range), "from")
+    to = DateFieldCollection.new(params.require(:stat_range), "to")
+    @model = StatRange.new(from:, to:)
+
+    return render :new unless @model.valid?
+
+    @collection = Cda::LinkingStatCollection.find_from_range(@model.from.to_s, @model.to.to_s)
+
+    render :show
```

Comment:
nit:
Instead of a clause guard, just a boring Rails pattern here?
```
if @model.valid?
   @collection = Cda::LinkingStatCollection.find_from_range(@model.from.to_s, @model.to.to_s)
    render :show
else
  render :new
end
```